cli.go

```diff
@@ -0,0 +1,387 @@
+	getSubscribedShardReq := &pbrpc.RPCGetSubscribedShardRequest{}
+	logger.Debugf("rpcclient:GetSubscribedShard: sending=%v", getSubscribedShardReq)
+	if res, err := client.GetSubscribedShard(context.Background(), getSubscribedShardReq); err != nil {
+		errMsg := fmt.Sprintf("Failed to get subscribed shards, err: %v", err)
+		emitCLIFailure(errMsg)
+	} else {
+		logger.Debugf("rpcclient:GetSubscribedShard: result=%v", res.ShardIDs)
+		shardResp := &ShardResponse{
+			ShardIDs: res.ShardIDs,
+		}
+		emitCLIResponse(shardResp)
+	}
+}
+
+func callRPCBroadcastCollation(
+	rpcAddr string,
+	shardID ShardIDType,
+	numCollations int,
+	collationSize int,
+	period int) {
+	conn, err := grpc.Dial(rpcAddr, grpc.WithInsecure())
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to connect to RPC server at %v, err: %v", rpcAddr, err)
+		emitCLIFailure(errMsg)
+	}
+	defer conn.Close()
+	client := pbrpc.NewPocClient(conn)
+	broadcastCollationReq := &pbrpc.RPCBroadcastCollationRequest{
+		ShardID: shardID,
+		Number:  PBInt(numCollations),
+		Size:    PBInt(collationSize),
+		Period:  PBInt(period),
+	}
+	logger.Debugf("rpcclient:BroadcastCollation: sending=%v", broadcastCollationReq)
+	if res, err := client.BroadcastCollation(context.Background(), broadcastCollationReq); err != nil {
+		errMsg := fmt.Sprintf(
+			"Failed to broadcast %v collations of size %v in period %v in shard %v, err: %v",
+			numCollations,
+			collationSize,
+			period,
+			shardID,
+			err,
+		)
+		emitCLIFailure(errMsg)
+	} else {
+		logger.Debugf("rpcclient:BroadcastCollation: result=%v", res)
+	}
+	emitCLIResponse(emptyResponse)
+}
+
+func callRPCStopServer(rpcAddr string) {
+	conn, err := grpc.Dial(rpcAddr, grpc.WithInsecure())
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to connect to RPC server at %v, err: %v", rpcAddr, err)
+		emitCLIFailure(errMsg)
+	}
+	defer conn.Close()
+	client := pbrpc.NewPocClient(conn)
+	stopServerReq := &pbrpc.RPCStopServerRequest{}
+	logger.Debugf("rpcclient:StopServer: sending=%v", stopServerReq)
+	if res, err := client.StopServer(context.Background(), stopServerReq); err != nil {
+		errMsg := fmt.Sprintf("Failed to stop RPC server at %v, err: %v", rpcAddr, err)
+		emitCLIFailure(errMsg)
+	} else {
+		logger.Debugf("rpcclient:StopServer: result=%v", res)
+	}
+	emitCLIResponse(emptyResponse)
+}
+
+func callRPCListPeer(rpcAddr string) {
+	conn, err := grpc.Dial(rpcAddr, grpc.WithInsecure())
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to connect to RPC server at %v, err: %v", rpcAddr, err)
+		emitCLIFailure(errMsg)
+	}
+	defer conn.Close()
+	client := pbrpc.NewPocClient(conn)
+	listPeerReq := &pbrpc.RPCListPeerRequest{}
+	logger.Debugf("rpcclient:ListPeer: sending=%v", listPeerReq)
+	res, err := client.ListPeer(context.Background(), listPeerReq)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to call RPC ListPeer at %v, err: %v", rpcAddr, err)
+		emitCLIFailure(errMsg)
+	}
+	peers := &PeerResponse{
+		Peers: res.Peers.Peers,
```

Comment:
`res.Peers` seems fine for the list of peers returned

Reply:
`res.Peers.Peers` is the `[]string`. `[]string` is needed for JSON marshaling